src/grt/src/fastroute/src/maze3D.cpp

```diff
@@ -395,6 +395,11 @@ int FastRouteCore::copyGrids3D(std::vector<TreeNode>& treenodes,
   const int n1x = treenodes[n1].x;
   const int n1y = treenodes[n1].y;
   const int n1l = treenodes[n1].botL;
+  const int routelen = treeedges[edge_n1n2].route.routelen;
+
+  gridsX_n1n2.reserve(routelen + 1);
```

Comment:
This is of course a performance optimization only and it doesn't have to be particularly precise... .resize() sets the size of the vector .reserve() only speeds up the reallocation by making that much space available for push_back & resize() later...